mp/src/game/server/filters.cpp

```diff
@@ -324,6 +326,40 @@ BEGIN_DATADESC( FilterTeam )
 END_DATADESC()
 
 
+// ###################################################################
+//	> FilterNeoClass
+// ###################################################################
+class FilterNeoClass : public CBaseFilter
+{
+	DECLARE_CLASS(FilterNeoClass, CBaseFilter);
+	DECLARE_DATADESC();
+
+public:
+	int m_iFilterNeoClass;
+
+	bool PassesFilterImpl(CBaseEntity* pCaller, CBaseEntity* pEntity)
+	{
+		if (!pEntity || !pEntity->IsPlayer())
+			return false;
+
+		CNEO_Player* pPlayer = dynamic_cast<CNEO_Player*>(pEntity);
```

Comment:
Since the only kind of player we have is `CNEO_Player`, and we've already passed `pEntity->IsPlayer` to get here, we can safely assume this must be a `CNEO_Player`. Therefore, this can be a `static_cast` for performance. Or you may use the `assert_cast`, which will assert the `dynamic_cast` for debug builds, but optimize it into `static_cast` for release builds, if you want the guarantee nonetheless.